ballerina/client.bal

```diff
@@ -256,6 +256,72 @@ public isolated client class Client {
         http:Response httpResponse = check self.amazonS3->delete(requestURI, request);
         return handleHttpResponse(httpResponse);
     }
+    
+    # Generates a presigned URL for the object.
+    #
+    # + bucketName - The name of the bucket  
+    # + objectName - The name of the object  
+    # + expirationTime - The time until the presigned URL is valid  
+    # + httpMethod - The HTTP method to be used, either GET or PUT  
+    # + partNo - The part number of the object, when uploading multipart objects
+    # + uploadId - The upload ID of the multipart upload
+    # + return - If success, a presigned URL, else an error
+    @display {label: "Create Presigned URL"}
+    remote isolated function createPresignedURL(@display {label: "Bucket Name"} string bucketName,
+    @display {label: "Object Name"} string objectName,
+    @display {label: "Expiration Time"} string expirationTime, 
+    @display {label: "HTTP Method"} string httpMethod, 
+    @display {label: "Part Number"} int partNo=0,
+    @display {label: "Upload ID"} string uploadId="") returns string|error? {
+
+        [string, string] [amzDateStr, shortDateStr] = ["", ""];
+        var result = generateDateString();
+        if result is [string, string] {
+            [amzDateStr, shortDateStr] = result;
+        } else {
+            return error("Error occurred while generating date string", result);
+        }
+        map<string> queryParams = {
+            ["X-Amz-Algorithm"]: AWS4_HMAC_SHA256,
+            ["X-Amz-Credential"]: self.accessKeyId + "/" + shortDateStr + "/" + self.region + "/" + SERVICE_NAME + "/" + TERMINATION_STRING,
+            ["X-Amz-Date"]: amzDateStr,
+            ["X-Amz-Expires"]: expirationTime,
+            ["X-Amz-SignedHeaders"]: "host"
```

Comment:
We don't need computed keys with literals.

I would actually introduce constants for these keys and then use computed keys.

Reply:
Can you explain this?

Reply:
Either

```ballerina
"X-Amz-Algorithm" : AWS4_HMAC_SHA256,
```

or (better approach)

```ballerina
const X_AMZ_ALGORITHM = "X-Amz-Algorithm"; // at module-level

...

[X_AMZ_ALGORITHM] : AWS4_HMAC_SHA256,
```